File: 006/Problem6.mjs
import { performance } from "perf_hooks";

function SumOfSquares(num) {
  var sum = 0;
  for (var i = 0; i < num + 1; i++) {
    sum += i * i;
  }
  return sum;
}

function SquareOfSum(num) {
  var sum = 0;
  for (var i = 0; i < num + 1; i++) {
    sum += i;
  }
  return sum * sum;
}

function problem() {
  console.log(SquareOfSum(100) - SumOfSquares(100));
}

var start = performance.now();
problem();
var end = performance.now();
console.log("Function took " + (end - start) / 1000);

File: 006/Problem6.rs
fn main() {
    let start = std::time::SystemTime::now();
    problem();
    match start.elapsed() {
        Ok(elapsed) => {
            println!("Function took: {}", elapsed.as_nanos())
        }
        Err(e) => {
            println!("Error: {:?}", e)
        }
    }
}

fn sum_of_squares(mut num: i64) -> i64 {
    let mut sum = 0;
    while num > 0 {
        sum += num * num;
        num -= 1;
    }
    return sum;
}

fn square_of_sum(mut num: i64) -> i64 {
    let mut sum = 0;
    while num > 0 {
        sum += num;
        num -= 1;
    }
    return sum * sum;
}

fn problem() {
    let natural_num = 100;
    println!("{}", square_of_sum(natural_num) - sum_of_squares(natural_num))
}
